### PR TITLE
ER-21 Backend - Implement Equipment Update API

### PR DESCRIPTION
**Pull request by YiJiePV**
```
/**
 * PUT/UPDATE EQUIPMENT
 * URL => /api/inventory/equipment/{serialId}
 * requestBody:
 *  {
 *      "schoolId": string,
 *      "typeId": int (optional),
 *      "modelId": int (optional),
 *      "maintenanceStatus": string (optional),
 *      "reservationStatus": string (optional),
 *      "usageCondition": string (optional),
 *      "purchaseCost": double (optional),
 *      "purchaseDate": date (optional)
 *  }
 * 
 * @return 404 or 400 if failed validation
 * @return 503 if server error
 * @return 200 OK 
 * {
 *      "message":"Equipment updated successfully."
 * }
 */
```

I implemented the Equipment Update API and created and used a Postman test suite to run regression E2E tests on it.
Similar to before, if you want to test this API with the test suite, start up the server with `npm start`, then locate and run each test request in the `ETC_API/Inventory/Update_Equipment_Tests` directory. You might have to change the `typeId`, `modelId`, and/or `serialId` parameters, since the items associated with them may have been deleted from the database ('cause they were bogus stand-ins for testing purposes 😜). You can change the values however you wish; feel free to try all kinds of things to break this API 😉.
And again, because I'm not a professional QA engineer, there's probably stuff I've missed even with all my test cases, so if you find any, please let me know! Thanks!

### CODE REVIEW

File: src_code/src/routes/InventoryRoutes.js
Comment (mohammedking11-spu) [inline src_code/src/routes/InventoryRoutes.js:1]: Thanks for including the requested body and how easy it is to follow the code. So far, it looks great in my opinion.

File: src_code/src/utils/services/helper_functions/general_helper_functions/GeneralHelperFunctions.js
Comment (mohammedking11-spu) [inline src_code/src/utils/services/helper_functions/general_helper_functions/GeneralHelperFunctions.js:1]: I liked how organized the code was. Great comments where they were not too lengthy, and not too vague, good job 😁

**Review by mohammedking11-spu — APPROVED** [review]
Hi Karena, your PR works as per expected! I tested the tests that you have made in Postman, and they seem to work very effectively. Everything seems to be organized (unless if I missed something, then I apologize.), and it follows the code convention. Code was straightforward and to the point, with no unnecessary stuff. Great job Karena, and I approve of this PR.

**Review by amansiid — APPROVED** [review]
Hello Karena, I have reviewed your code thoroughly, going through it function by function, and found it easy to follow. I've tested the Postman preset you provided, and it works as expected, catching wrong inputs for typeId and modelId. Just one extra line on your EquipmentUpdate,js line 198 and you've got a //TODO comment on line 347. Overall everything looks good, keep up the great work. I'll go ahead and approve this PR since it satisfied the main functionality. Thanks!

mysql result
postman test

**Pull request merged**